web3/_utils/module_testing/eth_module.py

```diff
@@ -202,6 +203,107 @@ def test_eth_sign(self, web3, unlocked_account_dual_type):
+                },
+                "message": {
+                    "from": {
+                        "name": "Cow",
+                        "wallet": "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"
+                    },
+                    "to": {
+                        "name": "Bob",
+                        "wallet": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"
+                    },
+                    "contents": "Hello, Bob!"
+                }
+            }
+        '''
+        signature = HexBytes(web3.eth.signTypedData(
+            unlocked_account_dual_type,
+            json.loads(validJSONMessage)
+        ))
+        assert len(signature) == 32 + 32 + 1
+
+    def test_invalid_eth_signTypedData(self,
+                                       web3,
+                                       unlocked_account_dual_type):
+        invalid_typed_message = '''
+            {
+                "types": {
+                    "EIP712Domain": [
+                        {"name": "name", "type": "string"},
+                        {"name": "version", "type": "string"},
+                        {"name": "chainId", "type": "uint256"},
+                        {"name": "verifyingContract", "type": "address"}
+                    ],
+                    "Person": [
+                        {"name": "name", "type": "string"},
+                        {"name": "wallet", "type": "address"}
+                    ],
+                    "Mail": [
+                        {"name": "from", "type": "Person"},
+                        {"name": "to", "type": "Person[2]"},
+                        {"name": "contents", "type": "string"}
+                    ]
+                },
+                "primaryType": "Mail",
+                "domain": {
+                    "name": "Ether Mail",
+                    "version": "1",
+                    "chainId": "0x01",
+                    "verifyingContract": "0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC"
+                },
+                "message": {
+                    "from": {
+                        "name": "Cow",
+                        "wallet": "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"
+                    },
+                    "to": [{
+                        "name": "Bob",
+                        "wallet": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"
+                    }],
+                    "contents": "Hello, Bob!"
+                }
+            }
+        '''
+        # ValueError is caused by the error in the response which is as follows
+        # ``Expected 2 items for array type Person[2], got 1 items``
+        with pytest.raises(ValueError) as e:
+            web3.eth.signTypedData(
+                unlocked_account_dual_type,
+                json.loads(invalid_typed_message)
+            )
+        try:
+            assert "Expected 2 items for array type Person[2], got 1 items" in str(e.value)
+        except AssertionError:
```

Comment:
Since this is getting `xfail`ed anyway, I don't think you need this `try`/`except` here?

Reply:
@kclowes I thought the same too, but it is still being failed. I wasn't sure why this is happening and hence resorted to this method.

Reply:
In that case, there is a method called `skip_if_testrpc` that I think would be better to use instead.